interfaces/sourcegen/pyproject.toml

```
@@ -0,0 +1,20 @@
+[build-system]
+requires = ["setuptools>=45", "wheel"]
+build-backend = "setuptools.build_meta"
```

Suggest using `hatchling` which is a much simpler build backend than setuptools

```suggestion
[build-system]
requires = ["hatchling"]
build-backend = "hatchling.build"
```